controllers/mirroring/mirroring_controller.go

```diff
@@ -209,13 +209,16 @@ func (r *MirroringReconciler) reconcilePhases(clientMappingConfig *corev1.Config
 		return ctrl.Result{RequeueAfter: 3 * time.Second}, nil
 	}
 
+	errorOccurred := false
+
 	ocsClient, err := providerClient.NewProviderClient(r.ctx, storageClusterPeer.Spec.ApiEndpoint, util.OcsClientTimeout)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to create a new provider client: %v", err)
+		r.log.Error(err, "failed to create a new provider client")
```

Comment:
if grpcClient isn't required b/n when it's created and used why can't the creation be moved to usage site?

Reply:
agreed, I will move creation to usage site